modules/drivers/athena/src/metabase/driver/athena.clj

```diff
@@ -315,58 +315,50 @@
     (catch Exception e
       (log/error (u/format-color 'red "Failed to execute query: %s %s" query (.getMessage e))))))
 
-(defn- describe-database->clj
-  "Workaround for wrong getColumnCount response by the driver (huh?)"
-  [column-metadata]
-  {:name (str/trim (:col_name column-metadata))
-   :type (str/trim (:data_type column-metadata))})
-
-(defn- remove-invalid-columns
-  "Returns a transducer."
-  []
-  (comp (remove #(= (:col_name %) ""))
-        (remove #(= (:col_name %) nil))
-        (remove #(= (:data_type %) nil))
-        (remove #(str/starts-with? (:col_name %) "#")) ; remove comment
-        (distinct)                                     ; driver can return twice the partitioning fields
-        (map describe-database->clj)))
+(defn- valid-column? [column-name database-type]
```

Comment:
I reworked the Athena sync code to be nice and efficient e.g. filter stuff out right away rather than fetch EVERYTHING and then filter stuff.